pkgdb/src/buildenv/realise.cc

```diff
@@ -238,155 +206,404 @@ getLockedPackages( resolver::Lockfile & lockfile, const System & system )
+                                          missingOutput ) );
+    }
+  auto outputsToOutpaths
+    = std::get<std::unordered_map<std::string, std::string>>(
+      maybeOutputsToOutpaths );
+  return outputsToOutpaths;
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+std::vector<std::pair<buildenv::RealisedPackage, nix::StorePath>>
+collectRealisedPackages(
+  nix::ref<nix::EvalState> &                     state,
+  const std::string &                            packageName,
+  const flox::resolver::LockedPackageRaw &       lockedPackage,
+  const std::string &                            parentOutpath,
+  std::unordered_map<std::string, std::string> & outputsToOutpaths )
+{
+  std::vector<std::pair<buildenv::RealisedPackage, nix::StorePath>> pkgs;
+  auto internalPriority = 0;
+  for ( const auto & [name, outpathStr] : outputsToOutpaths )
+    {
+      debugLog( "processing output '" + name + "' of '" + packageName + "'" );
+      auto outpathForOutput = state->store->parseStorePath( outpathStr );
+      buildenv::RealisedPackage pkg(
+        state->store->printStorePath( outpathForOutput ),
         true,
-        buildenv::Priority( package.priority,
-                            packagePath,
-                            /* idx should always fit in uint its unlikely a
-                             * package has more than 4 billion outputs. */
-                            static_cast<unsigned>( idx ) ) );
-      realised.emplace_back( realisedPackage, output.second.value() );
+        buildenv::Priority( lockedPackage.priority,
+                            parentOutpath,
+                            internalPriority++ ) );
+      pkgs.emplace_back( pkg, outpathForOutput );
     }
+  return pkgs;
+}
 
-  /* Build the derivation */
-  if ( auto drvPath = package_drv->queryDrvPath() )
+
+/* -------------------------------------------------------------------------- */
+
+std::vector<std::pair<buildenv::RealisedPackage, nix::StorePath>>
+getRealisedPackages( nix::ref<nix::EvalState> &         state,
+                     const std::string &                packageName,
+                     const resolver::LockedPackageRaw & lockedPackage,
+                     const System &                     system )
+{
+  auto timeEvalStart = std::chrono::high_resolution_clock::now();
+  auto cursor        = evalCacheCursorForInput( state,
+                                         lockedPackage.input,
+                                         lockedPackage.attrPath );
+
+  /* Try to eval the outPath. Trying this eval tells us whether the package is
+   * unsupported. This eval will fail in a number of cases:
+   * - The package doesn't work on this system
+   * - The package is marked "insecure" i.e. it's old (e.g. Python 2)
+   * - Possibly other cases as well
+   * */
+
+  // uses the cached value
+  auto parentOutpath
+    = tryEvaluatePackageOutPath( state, packageName, system, cursor );
+
+  // auto parentOutpath
```

Comment:
nb - Vestigial commented code?

Reply:
Yeah, I'll fix that one in the follow up.